semihost/machine/hexagon/hexagon_open.c

```diff
@@ -0,0 +1,50 @@
+#include "fcntl.h"
+#include "hexagon_semihost.h"
+#include <errno.h>
+#include <string.h>
+
+#define O_BINARY 0
+
+#define MODEMAP_LENGTH 14
+
+static const int modemap[MODEMAP_LENGTH] = {
```

Comment:
Given that the indices are going to be passed as the semimode, I'd love to see this documented in the code by creating #define values (or an enum) for each semimode and then using those to initialize the relevant array index. For instance:
```
#define HEX_SEMIMODE_READONLY_TEXT 0
...
    [HEX_SEMIMODE_READONLY_TEXT] = O_RDONLY,
```
This wouldn't change the generated code at all, but it would make the process a lot more clear.